src/System.Linq.Parallel/src/System/Linq/Parallel/Utils/ExceptionAggregator.cs

```diff
@@ -116,13 +115,16 @@ internal static void ThrowOCEorAggregateException(Exception ex, CancellationStat
                 CancellationState.ThrowWithStandardMessageIfCanceled(
                     cancellationState.ExternalCancellationToken);
             }
+            else if (ex is AggregateException)
+            {
+                throw ((AggregateException)ex).Flatten();
+            }
```

Comment:
I can't remember off hand, but I assume so.I didn't know if it was by-design, or by-accident.  Nothing else throws nested AEs, so I was assuming bug.  I can definitely pull it for more discussion in a later PR/issue or something.